src/Servers/Kestrel/Core/src/ListenOptionsHttpsExtensions.cs

```diff
@@ -179,6 +179,34 @@ public static ListenOptions UseHttps(this ListenOptions listenOptions, Action<Ht
         return listenOptions.UseHttps(options);
     }
 
+    /// <summary>
+    /// Retrieve the server certificate, if any, for an endpoint; otherwise, throw.
+    /// Should not be called before the configuration is loaded, if there is one.
+    /// </summary>
+    /// <param name="listenOptions">The <see cref="ListenOptions"/> to configure.</param>
+    /// <returns>The server certificate for this endpoint.</returns>
+    /// <exception cref="InvalidOperationException">If there is no server certificate for this endpoint.</exception>
+    /// <exception cref="InvalidOperationException">If there is a configuration and it has not been loaded.</exception>
+    public static X509Certificate2 GetServerCertificate(this ListenOptions listenOptions)
```

Comment:
What is the benefit of doing this as an extension method vs. just a regular method?

Reply:
Never mind I see its following a pattern here.

Reply:
I think we generally err on the side of making it an extension method if it doesn't require access to private state (for encapsulation?).